### PR TITLE
Bracket validation

### PR DESCRIPTION
**Pull request by MichaelJahns**
- [x] Bracket Validation and Tests
- [x] Update Master Readme
- [x] Create Local Readme
- [x] Whiteboard Image

### CODE REVIEW

File: java/src/test/java/testBracketValidation/testBracketValidation.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+    @Test
+    public void testEdgecases() {
+        String lone1 = ")";
+        String lone2 = "}";
+        String lone3 = "]";
+        assertTrue("Unexpected False", bracketValidation(lone1));
+        assertTrue("Unexpected False", bracketValidation(lone2));
+        assertTrue("Unexpected False", bracketValidation(lone3));
+        // I think my script is incorrect to return these as true. Sticking with
```
Comment (JessLovell) [inline java/src/test/java/testBracketValidation/testBracketValidation.java]: Yes. It is incorrect to return true for this.

File: java/src/test/java/testBracketValidation/testBracketValidation.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        // they all do technically have a pair, if this was a linter those
+        // would be classified as unpaired.
+
+        String multiple1 = "(((";
+        String multiple2 = "{{{";
+        String multiple3 = "[[[";
+        assertFalse("Unexpected True", bracketValidation(multiple1));
+        assertFalse("Unexpected True", bracketValidation(multiple2));
+        assertFalse("Unexpected True", bracketValidation(multiple3));
+        // I think my script is correct ot return these as false. I listed it as an edge
```
Comment (JessLovell) [inline java/src/test/java/testBracketValidation/testBracketValidation.java]: I agree here.

File: java/src/test/java/testBracketValidation/testBracketValidation.java [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+        assertTrue("Unexpected False", bracketValidation(Nonsense1));
+        assertTrue("Unexpected False", bracketValidation(Nonsense2));
+        assertTrue("Unexpected False", bracketValidation(Nonsense3));
+        assertFalse("Unexpected True", bracketValidation(Nonsense4));
+        assertFalse("Unexpected True", bracketValidation(Nonsense5));
+        assertFalse("Unexpected True", bracketValidation(Nonsense6));
+    }
+
+    @Test
+    public void testEdgecases() {
```
Comment (JessLovell) [inline java/src/test/java/testBracketValidation/testBracketValidation.java]: The instructions for this lab has some good edge cases to consider.

File: java/src/test/java/testBracketValidation/testBracketValidation.java [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+        // the below linter theme, these would be picked up and the return would be false
+        // I think theoretically I could add that functionality
+
+        String inverse1 = ")(";
+        String inverse2 = "][";
+        String inverse3 = "}{";
+        assertFalse("Unexpected True", bracketValidation(inverse1));
+        assertFalse("Unexpected True", bracketValidation(inverse2));
+        assertFalse("Unexpected True", bracketValidation(inverse3));
+        // I think that my script is correct to return these as false. Even
```
Comment (JessLovell) [inline java/src/test/java/testBracketValidation/testBracketValidation.java]: I agree with you here.

File: java/src/main/java/bracketValidation/bracketValidation.java [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+import java.util.HashSet;
+
+public class bracketValidation {
+    public static void main(String args[]) {
+        bracketValidation("()");
+    }
+
+    public static boolean bracketValidation(String input) {
+        HashSet<Character> h = new HashSet<>();
```
Comment (JessLovell) [inline java/src/main/java/bracketValidation/bracketValidation.java]: Use a stack instead of a hashset. This will fix some of the weird bugs you've got.
